fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/ResourceManagerHelperImpl.java

```diff
@@ -120,8 +123,17 @@ public ResourcePack open(String name) {
 
 					@Override
 					public ResourcePack openWithOverlays(String string, ResourcePackProfile.Metadata metadata) {
-						// Don't support overlays in builtin res packs.
-						return entry.getRight();
+						ModNioResourcePack pack = entry.getRight();
+						if (metadata.overlays().isEmpty()) {
+							return pack;
+						} else {
```

Comment:
Code style nit, no need for the else block, the code is slightly easier to read without it. Also make sure that checkstyle passes. I reccomend building using Gradle and/or using an IDE plugin for it.